File: solutions/1881. Maximum Value after Insertion/1881.py
class Solution:
  def maxValue(self, n: str, x: int) -> str:
    isNegative = n[0] == '-'

    for i, c in enumerate(n):
      if not isNegative and int(c) < x or isNegative and int(c) > x:
        return n[:i] + str(x) + n[i:]

    return n + str(x)

File: solutions/1881. Maximum Value after Insertion/1881.cpp
class Solution {
 public:
  string maxValue(string n, int x) {
    bool isNegative = n[0] == '-';

    for (int i = 0; i < n.length(); ++i)
      if (!isNegative && n[i] - '0' < x || isNegative && n[i] - '0' > x)
        return n.substr(0, i) + (char)('0' + x) + n.substr(i);

    return n + (char)('0' + x);
  }
};
